app/lib/database/driver/keyService.ts

```
@@ -0,0 +1,162 @@
+}
+
+// Per-storageKey in-flight map: serializes concurrent getOrCreate calls so only one
+// CSPRNG + setItem round-trip runs even when callers race.
+const _getOrCreateInflight = new Map<string, Promise<string>>();
+
+/**
+ * Generates or retrieves a hex material string for `storageKey`.
+ * Validates both stored values (corrupt → throw) and generated values (bad bridge → throw).
+ * Neither the stored value nor the generated value ever appears in thrown error messages.
+ */
+async function getOrCreate(sk: string, byteLen: number, hexLen: number, label: string): Promise<string> {
```

ESLint

[error] Async function 'getOrCreate' has no 'await' expression.

(require-await)